solidity/deploy/81_upgrade_wallet_coordinator_v2.ts

```diff
@@ -0,0 +1,98 @@
+import { Artifact, HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction, Deployment } from "hardhat-deploy/types"
+import { ContractFactory } from "ethers"
+import { ProxyAdmin, WalletCoordinator } from "../typechain"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { ethers, helpers, deployments } = hre
+
+  const { deployer } = await helpers.signers.getNamedSigners()
+
+  const proxyDeployment: Deployment = await deployments.get("WalletCoordinator")
+  const implementationContractFactory: ContractFactory =
+    await ethers.getContractFactory("WalletCoordinator", {
+      signer: deployer,
+    })
+
+  // Deploy new implementation contract
+  const newImplementationAddress: string = (await hre.upgrades.prepareUpgrade(
+    proxyDeployment,
+    implementationContractFactory,
+    {
+      kind: "transparent",
+    }
+  )) as string
+
+  deployments.log(
+    `new implementation contract deployed at: ${newImplementationAddress}`
+  )
+
+  // Assemble proxy upgrade transaction.
+  const proxyAdmin: ProxyAdmin = await hre.upgrades.admin.getInstance()
+  const proxyAdminOwner = await proxyAdmin.owner()
+
+  const upgradeTxData = await proxyAdmin.interface.encodeFunctionData(
+    "upgrade",
+    [proxyDeployment.address, newImplementationAddress]
+  )
+
+  deployments.log(
+    `proxy admin owner ${proxyAdminOwner} is required to upgrade proxy implementation with transaction:\n` +
+      `\t\tfrom: ${proxyAdminOwner}\n` +
+      `\t\tto: ${proxyAdmin.address}\n` +
+      `\t\tdata: ${upgradeTxData}`
+  )
+
+  // Assemble parameters upgrade transaction.
+  const walletCoordinator: WalletCoordinator =
+    await helpers.contracts.getContract("WalletCoordinator")
+
+  const walletCoordinatorOwner = await walletCoordinator.owner()
+
+  const updateRedemptionProposalParametersTxData =
+    walletCoordinator.interface.encodeFunctionData(
+      "updateRedemptionProposalParameters",
+      [7200, 600, 7200, 20, 20000]
+    )
+
+  deployments.log(
+    `WalletCoordinator owner ${walletCoordinatorOwner} is required to update redemption proposal parameters with transaction:\n` +
+      `\t\tfrom: ${walletCoordinatorOwner}\n` +
+      `\t\tto: ${walletCoordinator.address}\n` +
+      `\t\tdata: ${updateRedemptionProposalParametersTxData}`
+  )
```

Comment:
It is nice to be able to double-check the data in the simulation but worth noting, this part is not strictly required.

Reply:
Definitely, we will simulate with Tenderly first. But we need the new contract to be deployed for that.